django/thunderstore/frontend/tests/test_dynamic_html.py

```diff
@@ -0,0 +1,51 @@
+from typing import List
+
+import pytest
+
+from thunderstore.account.factories import UserFlagFactory
+from thunderstore.frontend.models import DynamicHTML, DynamicPlacement
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    ("required_flags", "excluded_flags", "user_flags", "should_match"),
+    [
+        ([], [], [], True),
+        ([], [], ["flag-1"], True),
+        (["flag-1"], [], [], False),
+        (["flag-1"], [], ["flag-1"], True),
+        (["flag-1"], [], ["flag-1", "flag-2"], True),
+        ([], ["flag-2"], [], True),
+        ([], ["flag-2"], ["flag-2"], False),
+        ([], ["flag-2"], ["flag-1", "flag-2"], False),
+        (["flag-1"], ["flag-2"], ["flag-1"], True),
+        (["flag-1"], ["flag-1", "flag-2"], ["flag-1"], False),
+        (["flag-1", "flag-2"], [], ["flag-2"], True),
```

Comment:
Seems like a possible footgun to allow permission by requiring any of the flags instead of all the required flags? 🤷🏻

Reply:
That's how it works for all of the other similar filters in the codebase right now, so it'd be more of a footgun to deviate from that IMO. I did add a help text to the field to clarify at least.